File: lib/screens/contact_screen.dart
import 'package:flutter/foundation.dart' show kIsWeb;
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:permission_handler/permission_handler.dart';
import 'package:url_launcher/url_launcher.dart';

import '../screens/auth/auth_manager.dart';
import '../data/app_contact.dart';
import '../data/contacts_cache.dart';
import '../data/contacts_sync_service.dart';
import '../data/local_contact_store.dart';
import '../screens/components/contact_action_sheet.dart';
import '../screens/components/contact_row.dart';
import '../screens/components/full_screen_profile_photo.dart';
import '../screens/chat/chat_helpers.dart';
import '../calls/call_initiator.dart';
import '../navigation/call_navigator.dart';

class ContactScreen extends StatefulWidget {
  const ContactScreen({super.key});

  @override
  State<ContactScreen> createState() => _ContactScreenState();
}

class _ContactScreenState extends State<ContactScreen> {
  // Tri-state, not bool: `status` (denied/granted/permanentlyDenied) drives
  // which banner (if any) renders above the list. The list itself never
  // waits on this — it always renders whatever's already in the local
  // Room-equivalent table (LocalContactStore), so a denied/revoked
  // permission degrades to chat-only contacts instead of an empty screen.
  PermissionStatus _permissionStatus = PermissionStatus.denied;
  bool _isSyncing = false;

  String? _currentUid;
  String _currentUserName = '';
  String _currentUserPhotoUrl = '';

  @override
  void initState() {
    super.initState();
    _loadCurrentUser();
    _loadFromLocalDbThenSync();
  }

  Future<void> _loadCurrentUser() async {
    await AuthManager.ensureSignedIn(
      onReady: () async {
        final cached = await AuthManager.getCachedProfile();
        if (cached == null || !mounted) return;
        setState(() {
          _currentUid = cached.uid;
          _currentUserName = cached.name;
          _currentUserPhotoUrl = cached.photoUrl;
        });
      },
      onError: (_) {},
    );
  }

  /// Instant paint from the local DB (no network, no permission prompt),
  /// then a background sync to refresh it — this is the "don't query
  /// ContactsContract/Firebase directly on every render" path.
  Future<void> _loadFromLocalDbThenSync() async {
    final cachedRows = await LocalContactStore.instance.getAll();
    if (mounted) ContactsCache.instance.setContacts(cachedRows);

    if (!kIsWeb) {
      final status = await Permission.contacts.status;
      if (mounted) setState(() => _permissionStatus = status);
    }

    await _syncInBackground();
  }

  Future<void> _syncInBackground() async {
    if (_currentUid == null) {
      final cached = await AuthManager.getCachedProfile();
      if (cached == null) return;
      _currentUid = cached.uid;
    }
    if (!mounted) return;
    setState(() => _isSyncing = true);
    await ContactsSyncService.sync(_currentUid!);
    ContactsCache.instance.hasLoadedOnce = true;
    if (mounted) setState(() => _isSyncing = false);
  }

  Future<void> _onPullToRefresh() => _syncInBackground();

  Future<void> _showPermissionRationale() async {
    if (kIsWeb) {
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(
          content: Text(
            "Contacts aren't available in the web version — open the app on your phone to see matched contacts.",
          ),
        ),
      );
      return;
    }

    if (_permissionStatus.isPermanentlyDenied) {
      final goToSettings = await showDialog<bool>(
        context: context,
        builder: (context) => AlertDialog(
          title: const Text('Contacts access is off'),
          content: const Text(
            "You previously denied contacts access. To see which of your contacts "
            "are using the app, turn on Contacts permission in your phone's app settings.",
          ),
          actions: [
            TextButton(onPressed: () => Navigator.pop(context, false), child: const Text('Not now')),
            TextButton(onPressed: () => Navigator.pop(context, true), child: const Text('Open Settings')),
          ],
        ),
      );
      if (goToSettings == true) await openAppSettings();
      return;
    }

    final proceed = await showDialog<bool>(
      context: context,
      builder: (context) => AlertDialog(
        title: const Text('Allow contacts access?'),
        content: const Text(
          "We'll check your contacts' phone numbers against people already using the app, "
          "so you can message them without adding anything manually. Your contacts are never "
          'uploaded — only phone numbers are compared, locally, against registered users.',
        ),
        actions: [
          TextButton(onPressed: () => Navigator.pop(context, false), child: const Text('Not now')),
          TextButton(onPressed: () => Navigator.pop(context, true), child: const Text('Continue')),
        ],
      ),
    );
    if (proceed != true) return;

    final status = await Permission.contacts.request();
    if (!mounted) return;
    setState(() => _permissionStatus = status);
    if (status.isGranted) {
      await _syncInBackground();
    }
  }

  void _showAddContactUnsupported() {
    ScaffoldMessenger.of(context).showSnackBar(
      const SnackBar(
        content: Text('Add a contact using your phone\'s Contacts app, then come back and tap Refresh.'),
      ),
    );
  }

  Future<void> _sendSms(String phone) async {
    final uri = Uri(scheme: 'sms', path: phone);
    if (await canLaunchUrl(uri)) {
      await launchUrl(uri);
    }
  }

  Future<void> _dialPhone(String phone) async {
    final uri = Uri(scheme: 'tel', path: phone);
    if (await canLaunchUrl(uri)) {
      await launchUrl(uri);
    }
  }

  void _showVideoCallUnavailable() {
    ScaffoldMessenger.of(context).showSnackBar(
      const SnackBar(content: Text('Video calling is only available for contacts on the app.')),
    );
  }

  // Mirrors _startCall in private_chat_screen.dart — same CallInitiator /
  // CallNavigator flow, just addressed at a contact's uid instead of a
  // chat's fixed otherUid.
  Future<void> _startCall(AppContact contact, String callType) async {
    final uid = _currentUid;
    final calleeUid = contact.uid;
    if (uid == null || calleeUid == null) return;

    final initiator = CallInitiator();
    await initiator.initiateCall(
      callerUid: uid,
      callerName: _currentUserName,
      callerPhoto: _currentUserPhotoUrl,
      calleeUid: calleeUid,
      chatId: privateChatId(uid, calleeUid),
      callType: callType,
      onCallIdReady: (callId) {
        CallNavigator.toOutgoingCall(
          callId: callId,
          calleeName: contact.name,
          callType: callType,
        );
      },
    );
  }

  void _openActionSheet(AppContact contact) {
    showContactActionSheet(
      context: context,
      contact: contact,
      onMessage: (c) {
        if (c.uid != null) {
          context.push('/privateChat/${c.uid}');
        } else {
          _sendSms(c.phone);
        }
      },
      onVoiceCall: (c) {
        if (c.uid != null) {
          _startCall(c, 'audio');
        } else {
          _dialPhone(c.phone);
        }
      },
      onVideoCall: (c) {
        if (c.uid != null) {
          _startCall(c, 'video');
        } else {
          _showVideoCallUnavailable();
        }
      },
    );
  }

  void _confirmClearContacts() {
    showDialog(
      context: context,
      builder: (context) => AlertDialog(
        title: const Text('Clear contact list?'),
        content: const Text(
          'This clears the list shown here. It won\'t delete anything from your phone\'s contacts. '
          'You can rebuild it anytime with Refresh.',
        ),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(context),
            child: const Text('Cancel'),
          ),
          TextButton(
            onPressed: () async {
              await LocalContactStore.instance.clearAll();
              ContactsCache.instance.setContacts([]);
              if (context.mounted) Navigator.pop(context);
            },
            child: const Text('Clear'),
          ),
        ],
      ),
    );
  }

  void _showDiscussionPicker() {
    final eligible = ContactsCache.instance.contacts.where((c) => c.uid != null).toList();
    showDialog(
      context: context,
      builder: (context) => AlertDialog(
        title: const Text('Start a discussion with…'),
        content: SizedBox(
          width: double.maxFinite,
          child: eligible.isEmpty
              ? const Text('No matched contacts yet. Try Refresh first.')
              : ConstrainedBox(
                  constraints: const BoxConstraints(maxHeight: 320),
                  child: ListView.builder(
                    shrinkWrap: true,
                    itemCount: eligible.length,
                    itemBuilder: (context, index) {
                      final contact = eligible[index];
                      return ListTile(
                        title: Text(contact.name),
                        onTap: () {
                          Navigator.pop(context);
                          context.push('/privateChat/${contact.uid}');
                        },
                      );
                    },
                  ),
                ),
        ),
        actions: [
          TextButton(onPressed: () => Navigator.pop(context), child: const Text('Cancel')),
        ],
      ),
    );
  }

  void _showPhoto(AppContact contact) {
    showGeneralDialog(
      context: context,
      barrierColor: Colors.transparent,
      barrierDismissible: false,
      pageBuilder: (context, __, ___) {
        return FullScreenProfilePhoto(
          name: contact.name,
          photoUrl: contact.photoUrl,
          isOwnProfile: contact.uid == _currentUid,
          onDismiss: () => Navigator.pop(context),
        );
      },
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Stack(
        children: [
          Container(
            decoration: const BoxDecoration(
              gradient: LinearGradient(
                begin: Alignment.topCenter,
                end: Alignment.bottomCenter,
                colors: [Color(0xFFF6F9FF), Color(0xFFE7F0FF), Colors.white],
              ),
            ),
            child: SafeArea(
              child: Padding(
                padding: const EdgeInsets.symmetric(horizontal: 22),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.center,
                  children: [
                    const SizedBox(height: 36),
                    const Text('WELCOME TO', style: TextStyle(fontSize: 15, color: Colors.grey, fontWeight: FontWeight.w500)),
                    const SizedBox(height: 8),
                    const Text(
                      'TONKOH BIBLE\nTRUTH REVEALERS',
                      textAlign: TextAlign.center,
                      style: TextStyle(fontSize: 28, fontWeight: FontWeight.w800, color: Color(0xFF102A72)),
                    ),
                    const SizedBox(height: 10),
                    const Text(
                      'We Preach Christ and Him Crucified',
                      textAlign: TextAlign.center,
                      style: TextStyle(fontSize: 14, color: Color(0xFFB71C1C), fontWeight: FontWeight.w600),
                    ),
                    const SizedBox(height: 24),
                    Align(
                      alignment: Alignment.centerLeft,
                      child: Padding(
                        padding: const EdgeInsets.only(bottom: 10),
                        child: Text('Contacts on the app',
                            style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold, color: Color(0xFF102A72))),
                      ),
                    ),
                    if (!kIsWeb && !_permissionStatus.isGranted) _buildPermissionBanner(),
                    Expanded(child: _buildBody()),
                  ],
                ),
              ),
            ),
          ),
          Positioned(
            top: 8,
            right: 8,
            child: SafeArea(
              child: PopupMenuButton<String>(
                icon: const Icon(Icons.more_vert, color: Color(0xFF102A72)),
                onSelected: (value) {
                  switch (value) {
                    case 'add':
                      _showAddContactUnsupported();
                      break;
                    case 'refresh':
                      _syncInBackground();
                      break;
                    case 'delete':
                      _confirmClearContacts();
                      break;
                    case 'discussion':
                      _showDiscussionPicker();
                      break;
                  }
                },
                itemBuilder: (context) => const [
                  PopupMenuItem(value: 'add', child: Text('Add contact')),
                  PopupMenuItem(value: 'refresh', child: Text('Refresh')),
                  PopupMenuItem(value: 'delete', child: Text('Delete contacts')),
                  PopupMenuItem(value: 'discussion', child: Text('Create a two person discussion')),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildPermissionBanner() {
    return Padding(
      padding: const EdgeInsets.only(bottom: 12),
      child: Card(
        color: const Color(0xFFEFF3FF),
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14)),
        elevation: 0,
        child: Padding(
          padding: const EdgeInsets.all(14),
          child: Row(
            children: [
              const Icon(Icons.contacts_outlined, color: Color(0xFF102A72)),
              const SizedBox(width: 10),
              const Expanded(
                child: Text(
                  'Allow contacts access to see more people you know on the app.',
                  style: TextStyle(fontSize: 12.5),
                ),
              ),
              TextButton(onPressed: _showPermissionRationale, child: const Text('Allow')),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildBody() {
    return RefreshIndicator(
      onRefresh: _onPullToRefresh,
      child: ListenableBuilder(
        listenable: ContactsCache.instance,
        builder: (context, _) {
          final contacts = ContactsCache.instance.contacts;
          if (contacts.isEmpty) {
            if (_isSyncing) {
              return const Center(child: CircularProgressIndicator());
            }
            return ListView(
              // ListView (not a bare Text) so pull-to-refresh still works
              // on an empty list — a Center/Text alone has no scrollable
              // surface for RefreshIndicator to attach its gesture to.
              children: const [
                Padding(
                  padding: EdgeInsets.symmetric(vertical: 30),
                  child: Text(
                    'None of your contacts are using the app yet.',
                    textAlign: TextAlign.center,
                    style: TextStyle(fontSize: 13, color: Colors.grey),
                  ),
                ),
              ],
            );
          }
          return ListView.separated(
            itemCount: contacts.length,
            separatorBuilder: (_, __) => const SizedBox(height: 10),
            itemBuilder: (context, index) {
              final contact = contacts[index];
              return ContactRow(
                contact: contact,
                onTap: () => _openActionSheet(contact),
                onAvatarTap: contact.uid != null ? () => _showPhoto(contact) : null,
              );
            },
          );
        },
      ),
    );
  }
}

File: android/app/src/main/kotlin/com/tbtrapp/screens/ContactScreen.kt
package com.tbtrapp.screens

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.provider.ContactsContract
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Call
import androidx.compose.material.icons.filled.MoreVert
import androidx.compose.material.icons.filled.Videocam
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.content.ContextCompat
import androidx.navigation.NavController
import com.tbtrapp.auth.AuthManager
import com.tbtrapp.calls.initiateCall
import com.tbtrapp.calls.OutgoingCallActivity
import com.tbtrapp.data.AppContact
import com.tbtrapp.data.ContactsCache
import com.tbtrapp.data.ContactsRepository
import com.tbtrapp.data.ProfilePhotoCache
import com.tbtrapp.components.ContactActionSheet
import com.tbtrapp.components.ContactRow
import com.tbtrapp.components.FullScreenProfilePhoto
import com.tbtrapp.util.PhoneActions
import kotlinx.coroutines.launch
import androidx.compose.foundation.shape.RoundedCornerShape


@Composable
fun ContactScreen(
    navController: NavController
) {
    val context = LocalContext.current
    val scope = rememberCoroutineScope()

    val background = Brush.verticalGradient(
        listOf(Color(0xFFF6F9FF), Color(0xFFE7F0FF), Color(0xFFFFFFFF))
    )

    var hasPermission by remember {
        mutableStateOf(
            ContextCompat.checkSelfPermission(
                context, Manifest.permission.READ_CONTACTS
            ) == PackageManager.PERMISSION_GRANTED
        )
    }

    val appContacts by ContactsCache.contacts

    var isLoading by remember { mutableStateOf(false) }

    var menuExpanded by remember { mutableStateOf(false) }
    var showDeleteConfirm by remember { mutableStateOf(false) }
    var showDiscussionPicker by remember { mutableStateOf(false) }

    var selectedContact by remember { mutableStateOf<AppContact?>(null) }
    var photoViewerContact by remember { mutableStateOf<AppContact?>(null) }

    // ── current user info needed for calling ──
    var currentUid by remember { mutableStateOf<String?>(null) }
    var currentUserName by remember { mutableStateOf("") }

    val permissionLauncher = rememberLauncherForActivityResult(
        contract = ActivityResultContracts.RequestPermission()
    ) { granted -> hasPermission = granted }

    suspend fun refreshContacts() {
        if (!hasPermission) return
        isLoading = true
        val fetched = ContactsRepository.fetchMatchedContacts(context)
        ContactsCache.contacts.value = fetched
        ContactsCache.hasLoadedOnce = true
        isLoading = false
    }

    suspend fun syncContactsQuietly() {
        if (!hasPermission) return
        val fetched = ContactsRepository.fetchMatchedContacts(context)
        ContactsCache.contacts.value = fetched
    }

    LaunchedEffect(Unit) {
        if (!hasPermission) permissionLauncher.launch(Manifest.permission.READ_CONTACTS)
    }

    LaunchedEffect(hasPermission) {
        if (!hasPermission) return@LaunchedEffect

        if (!ContactsCache.hasLoadedOnce) {
            refreshContacts()
        } else {
            syncContactsQuietly()
        }
    }

    // Fetch current user once so we can initiate calls
    LaunchedEffect(Unit) {
        val uid = AuthManager.currentUid(context)
        currentUid = uid
        if (uid != null) {
            AuthManager.fetchProfile(
                context = context,
                phoneKey = uid,
                onResult = { profile ->
                    currentUserName = profile?.name ?: ""
                }
            )
        }
    }

    Box(modifier = Modifier.fillMaxSize()) {

        Column(
            modifier = Modifier
                .fillMaxSize()
                .background(background)
                .padding(horizontal = 22.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {

            Spacer(modifier = Modifier.height(36.dp))

            Text("WELCOME TO", fontSize = 15.sp, color = Color.Gray, fontWeight = FontWeight.Medium)
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = "TONKOH BIBLE\nTRUTH REVEALERS",
                textAlign = TextAlign.Center,
                fontSize = 28.sp,
                fontWeight = FontWeight.ExtraBold,
                color = Color(0xFF102A72)
            )
            Spacer(modifier = Modifier.height(10.dp))
            Text(
                text = "We Preach Christ and Him Crucified",
                textAlign = TextAlign.Center,
                fontSize = 14.sp,
                color = Color(0xFFB71C1C),
                fontWeight = FontWeight.SemiBold
            )
            Spacer(modifier = Modifier.height(24.dp))

            Text(
                text = "Contacts on the app",
                fontSize = 16.sp,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF102A72),
                modifier = Modifier.fillMaxWidth().padding(bottom = 10.dp),
                textAlign = TextAlign.Start
            )

            when {
                !hasPermission -> {
                    Card(shape = RoundedCornerShape(18.dp), modifier = Modifier.fillMaxWidth()) {
                        Column(
                            modifier = Modifier.padding(20.dp),
                            horizontalAlignment = Alignment.CenterHorizontally
                        ) {
                            Text(
                                "Contact permission is needed to show which of your contacts are using the app.",
                                textAlign = TextAlign.Center,
                                fontSize = 13.sp
                            )
                            Spacer(modifier = Modifier.height(12.dp))
                            Button(onClick = {
                                permissionLauncher.launch(Manifest.permission.READ_CONTACTS)
                            }) { Text("Grant Access") }
                        }
                    }
                }

                isLoading -> {
                    Box(
                        modifier = Modifier.fillMaxWidth().padding(vertical = 40.dp),
                        contentAlignment = Alignment.Center
                    ) { CircularProgressIndicator() }
                }

                appContacts.isEmpty() -> {
                    Text(
                        "None of your contacts are using the app yet.",
                        fontSize = 13.sp,
                        color = Color.Gray,
                        modifier = Modifier.padding(vertical = 30.dp)
                    )
                }

                else -> {
                    LazyColumn(
                        modifier = Modifier.fillMaxWidth(),
                        verticalArrangement = Arrangement.spacedBy(10.dp)
                    ) {
                        items(appContacts) { contact ->
                            Box(modifier = Modifier.fillMaxWidth()) {
                                ContactRow(
                                    contact = contact,
                                    onClick = { selectedContact = contact },
                                    onAvatarClick = {
                                        contact.uid?.let { photoViewerContact = contact }
                                    }
                                )

                                // WhatsApp-style audio / video call buttons

                            }
                        }
                    }
                }
            }

            Spacer(modifier = Modifier.height(20.dp))
        }

        Box(
            modifier = Modifier
                .align(Alignment.TopEnd)
                .statusBarsPadding()
                .padding(top = 8.dp, end = 8.dp)
        ) {
            IconButton(onClick = { menuExpanded = true }) {
                Icon(Icons.Default.MoreVert, contentDescription = "More options", tint = Color(0xFF102A72))
            }

            DropdownMenu(expanded = menuExpanded, onDismissRequest = { menuExpanded = false }) {
                DropdownMenuItem(
                    text = { Text("Add contact") },
                    onClick = {
                        menuExpanded = false
                        try {
                            context.startActivity(
                                Intent(Intent.ACTION_INSERT).apply {
                                    type = ContactsContract.Contacts.CONTENT_TYPE
                                }
                            )
                        } catch (e: Exception) {
                            android.util.Log.e("ContactScreen", "No Contacts app found", e)
                        }
                    }
                )
                DropdownMenuItem(
                    text = { Text("Refresh") },
                    onClick = { menuExpanded = false; scope.launch { refreshContacts() } }
                )
                DropdownMenuItem(
                    text = { Text("Delete contacts") },
                    onClick = { menuExpanded = false; showDeleteConfirm = true }
                )
                DropdownMenuItem(
                    text = { Text("Create a two person discussion") },
                    onClick = { menuExpanded = false; showDiscussionPicker = true }
                )
            }
        }
    }

    selectedContact?.let { contact ->
        ContactActionSheet(
            contact = contact,
            onDismiss = { selectedContact = null },
            onMessage = { c ->
                if (c.uid != null) {
                    navController.navigate("privateChat/${c.uid}")
                } else {
                    PhoneActions.sendSms(context, c.phone)
                }
            },
            onVoiceCall = { c ->
                val uid = currentUid
                val calleeUid = c.uid
                if (uid != null && calleeUid != null) {
                    initiateCall(
                        callerUid = uid,
                        callerName = currentUserName,
                        calleeUid = calleeUid,
                        chatId = "",
                        callType = "audio",
                        onCallIdReady = { callId ->
                            context.startActivity(
                                OutgoingCallActivity.buildIntent(context, callId, c.name, "audio")
                            )
                        }
                    )
                }
            },
            onVideoCall = { c ->
                val uid = currentUid
                val calleeUid = c.uid
                if (uid != null && calleeUid != null) {
                    initiateCall(
                        callerUid = uid,
                        callerName = currentUserName,
                        calleeUid = calleeUid,
                        chatId = "",
                        callType = "video",
                        onCallIdReady = { callId ->
                            context.startActivity(
                                OutgoingCallActivity.buildIntent(context, callId, c.name, "video")
                            )
                        }
                    )
                }
            }
        )
    }

    photoViewerContact?.let { contact ->
        FullScreenProfilePhoto(
            name = contact.name,
            photoUrl = contact.uid?.let { ProfilePhotoCache.photoFor(context, it) } ?: "",
            isOwnProfile = contact.uid == AuthManager.currentUid(context),
            onDismiss = { photoViewerContact = null }
        )
    }

    if (showDeleteConfirm) {
        AlertDialog(
            onDismissRequest = { showDeleteConfirm = false },
            title = { Text("Clear contact list?") },
            text = {
                Text("This clears the list shown here. It won't delete anything from your phone's contacts. You can rebuild it anytime with Refresh.")
            },
            confirmButton = {
                TextButton(onClick = {
                    ContactsCache.contacts.value = emptyList()
                    showDeleteConfirm = false
                }) { Text("Clear") }
            },
            dismissButton = {
                TextButton(onClick = { showDeleteConfirm = false }) { Text("Cancel") }
            }
        )
    }

    if (showDiscussionPicker) {
        AlertDialog(
            onDismissRequest = { showDiscussionPicker = false },
            title = { Text("Start a discussion with…") },
            text = {
                val eligible = appContacts.filter { it.uid != null }
                if (eligible.isEmpty()) {
                    Text("No matched contacts yet. Try Refresh first.")
                } else {
                    LazyColumn(
                        modifier = Modifier.heightIn(max = 320.dp),
                        verticalArrangement = Arrangement.spacedBy(4.dp)
                    ) {
                        items(eligible) { contact ->
                            Text(
                                text = contact.name,
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .clickable {
                                        showDiscussionPicker = false
                                        navController.navigate("privateChat/${contact.uid}")
                                    }
                                    .padding(vertical = 10.dp)
                            )
                        }
                    }
                }
            },
            confirmButton = {
                TextButton(onClick = { showDiscussionPicker = false }) { Text("Cancel") }
            }
        )
    }
}